### ISSUE TITLE
containerized-gadgets: Support running gadgets from private registries in Kubernetes

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#2096]
We need to understand how to support pulling gadget images from private registries in the Kubernetes case. From a quick research, it seems we should use Kubernetes secrets and allow the run command to point to one of those. (as done in https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/ for pods)

```bash
$ kubectl create secret docker-registry regcred --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>

... 

$ kubectl gadget run ghcr.io/foo/bar --image-pull-secrets=regcred
```

It's annoying for users to specify the secrets each time, so we also need to think about a "default" secret that is used when no secret is specified.

**Comment by mqasimsarfraz** [comment]
How about handling it at deployment time:
```
$ kubectl gaget deploy --image-pull-secrets=regcred 
```
And when specified, we can just mount the credentials in the pod, [example](https://github.com/GoogleContainerTools/kaniko/tree/main#pushing-to-azure-container-registry).

**Comment by mauriciovasquezbernal** [comment]
Makes sense to me. But I think we should also allow users to specify it in the `run` command in case they want to use a given credentials.

**Comment by mqasimsarfraz** [comment]
Sounds good. I will pick this one up!
